primedev/squirrel/squirrel.cpp

```diff
@@ -207,9 +207,6 @@ template <ScriptContext context> void SquirrelManager<context>::VMCreated(CSquir
 	defconst(m_pSQVM, "NS_VERSION_PATCH", version[2]);
 	defconst(m_pSQVM, "NS_VERSION_DEV", version[3]);
 
-	// define squirrel constant for if we are in vanilla-compatibility mode
-	defconst(m_pSQVM, "VANILLA", g_pVanillaCompatibility->GetVanillaCompatibility());
```

Comment:
Why? This is an unnecessary breaking change, and disables the ability to disable compilation of non-vanilla stuff that might break vanilla compatibility.

Vanilla compatibility is defined at startup anyways and cannot be changed, no reason to do this.

Reply:
> Vanilla compatibility is defined at startup anyways and cannot be changed, no reason to do this.

Most of the changes in this PR are to dynamically switch around vanilla compat. You can't have any real compile-time checking for this, especially when most of the checks for it would be on uiscript. As such these breaking changes are nessecary